### DISCUSSION TITLE
[Feature] Trashed files location in file system

### DISCUSSION THREAD

**Discussion by bsmely-w** [immich-app/immich#5778] [Feature Request]
### The feature

It would be great to have a different location for storing trashed files in the file system.

Currently, when a media file is moved to the trash, it remains in the usual storage template location until it is completely removed. I would like to suggest moving trashed files to a ".trash" folder located in the root of the library. Within the ".trash" folder, the files may still be stored under the storage template structure.

For example:

```
Before trashing:
library/admin/2023/file-1.jpg
library/admin/2023/file-2.jpg (to be deleted)

After trashing:
library/admin/2023/file-1.jpg
library/admin/.trash/2023/file-2.jpg (deleted)
```

The main reason for this suggestion is to keep the library clear, especially in cases where users prefer a long trashing period before completely removing files.

Note:
In my current setup, the data flow is as follows:
Immich library -> Nextcloud (as read-only file system) -> read-only folder under Windows PC to have fast local access to the media files and as an additional library backup.
Keeping a lot of trashed objects mixed with "untrashed" files is not very useful.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bsmely-w** [comment]
To trying test this approach, I prepared small patch, but currently I have issues with building docker image and can't test it yet.

```
diff --git a/server/src/domain/storage-template/storage-template.service.ts b/server/src/domain/storage-template/storage-template.service.ts
index cbaf5541..ac28e8e4 100644
--- a/server/src/domain/storage-template/storage-template.service.ts
+++ b/server/src/domain/storage-template/storage-template.service.ts
@@ -94,7 +94,7 @@ export class StorageTemplateService {
   async handleMigration() {
     this.logger.log('Starting storage template migration');
     const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) =>
-      this.assetRepository.getAll(pagination),
+      this.assetRepository.getAll(pagination, {withDeleted: true}),
     );
     const users = await this.userRepository.getList();
 
@@ -147,9 +147,10 @@ export class StorageTemplateService {
 
     try {
       const source = asset.originalPath;
-      const ext = path.extname(source).split('.').pop() as string;
+      const ext = path.extname(source).split('.').pop()?.toLowerCase() as string;
       const sanitized = sanitize(path.basename(filename, `.${ext}`));
       const rootPath = StorageCore.getLibraryFolder({ id: asset.ownerId, storageLabel });
+      const typeDir = this.getAssetTypeDirectory(asset);
 
       let albumName = null;
       if (this.template.needsAlbum) {
@@ -163,7 +164,7 @@ export class StorageTemplateService {
         extension: ext,
         albumName,
       });
-      const fullPath = path.normalize(path.join(rootPath, storagePath));
+      const fullPath = path.normalize(path.join(rootPath, typeDir, storagePath));
       let destination = `${fullPath}.${ext}`;
 
       if (!fullPath.startsWith(rootPath)) {
@@ -216,6 +217,16 @@ export class StorageTemplateService {
     }
   }
 
+  private getAssetTypeDirectory(asset: AssetEntity) {
+    if (asset.deletedAt) {
+      return '.trash';
+    }
+    if (asset.isArchived) {
+      return '.archive';
+    }
+    return '';
+  }
+
   private validate(config: SystemConfig) {
     try {
       const { compiled } = this.compile(config.storageTemplate.template);
```

**Reply by AlleNPooh** [reply]
支持你的工作，已经都进入代码阶段了，祝你早日成功，这个功能很多人都是非常需要的。

**Comment by bsmely** [comment]
I created a fork with requested changes https://github.com/bsmely/immich/compare/v1.98.1...bsmely:immich:migration-improvements

**Reply by MrModest** [reply]
Oh, adding variables to the "Storage Template" looks even smarter! Now you can sort not only Trash, but Archive as well 🤩
